rhv_checks.py

```diff
@@ -338,6 +339,43 @@ def check_vms_distributed_hosts(system, warn=5, crit=10):
         sys.exit(3)
 
 
+def check_hosted_engine_status(system, **kwargs):
+    """ Check the status of all the host's Hosted Engine Status."""
+    okay, critical, warning = [], [], []
+    # Get all the hosts with details.
+    hosts = system.api.system_service().hosts_service().list(all_content=True)
+
+    for host in hosts:
+        engine_value = all((host.hosted_engine.active, host.hosted_engine.configured))
+        maintenance_value = any((host.hosted_engine.global_maintenance, host.hosted_engine.local_maintenance))
+        if not engine_value or maintenance_value:
+            critical.append((host.name, "False"))
```

Comment:
Why not append the boolean or return the whole hosted engine status so that we can tell what the problem is? I.e. is it the `engine_value` or the `maintenance_value` that's causing problems?

Reply:
I'd say let's return dictionary(you can create this dictionary before `engine_value` and `maintenance_value` , that way we not only get name of the faulty host, but also get all the info and can quickly tell what's wrong: 
```
host_info_dict =  {'Configured': host.hosted_engine.configured,  'Active': host.hosted_engine.active,
'local_maintenance': host.hosted_engine.local_maintenance, 'global_maintenance': host.hosted_engine.global_maintenance }

critical.append((host.name, host_info_dict))

```

You can then reference same dictionary  even in `engine_value` and `maintenance_value`

Reply:
> Why not append the boolean or return the whole hosted engine status so that we can tell what the problem is? I.e. is it the `engine_value` or the `maintenance_value` that's causing problems?

We are not sure about it, that's what returned final output.

Reply:
> I'd say let's return dictionary(you can create this dictionary before `engine_value` and `maintenance_value` , that way we not only get name of the faulty host, but also get all the info and can quickly tell what's wrong:
> 
> ```
> host_info_dict =  {'Configured': host.hosted_engine.configured,  'Active': host.hosted_engine.active,
> 'local_maintenance': host.hosted_engine.local_maintenance, 'global_maintenance': host.hosted_engine.global_maintenance }
> 
> critical.append((host.name, host_info_dict))
> ```
> 
> You can then reference same dictionary even in `engine_value` and `maintenance_value`

Okay, I will give a try and update accordingly.